### PR TITLE
Changes to antidote CRDTs (as per M12)

### PR DESCRIPTION
**Pull request by gyounes**
Changes done:
- remove "antidote_crdt" prefix from names
- rename crdts to match pattern: <Type>_<semantics>
- move 3 data types to experimental branch: integer, rga, map_aw
- common flag code -> flag_helper
- remove map.erl
- some refactoring

Additional change:
- I removed the reset operation from map_go as it has a grow-only semantics.

### CODE REVIEW

**Comment by peterzeller** [comment]
Why did you rename all modules? I thought it was common practice in Erlang to prefix module names with some common name to avoid conflicts with other libraries. 

Unfortunately, the renaming also makes the diff hard to read, so I would at least move the renaming to a different pull request.

Also: `TEST-file_antidote_crdt.app.xml` should not be in the repository.

**Comment by gyounes** [comment]
Hi Peter,
Those were decisions made in an antidote CRDTs meeting during the M12... @bieniusa 
Regarding the ``TEST-file_antidote_crdt.app.xml``: agreed!

**Comment by bieniusa** [comment]
I already updated the .gitignore regarding the test xml files.
The renaming was discussed and decided at Lightkone M12. 

We should pull #18 first, though, I guess?

**Comment by cmeiklejohn** [comment]
Peter Zeller is right about the prefix being a common practice, because of the lack of namespacing in Erlang, and if the goal was to make it easier from a usability point of view, the most common way that's done is to have something that maps user-names to underlying modules.  

What was the motivation behind the request?

**Comment by bieniusa** [comment]
The motivation is a general cleanup of this library.

I had a look at the Lasp CRDTs - there is also no common prefix.
So, I assumed that it is indeed not necessary?
The problem is that the names get really long.

**Comment by cmeiklejohn** [comment]
`state_orset.erl` (from the Lasp library) is quite different from having `map.erl`

But, as you wish.  

In retrospect, we probably should have prefixed all of them with `types_` and if you look at every other Lasp library we use the prefix: `partisan`, `sprinter`, and `lasp` itself all use the prefix.

**Comment by bieniusa** [comment]
The map.erl should be gone; we will have the map_rr and a map_aw.
But I don't understand what you are suggesting:
- to keep the antidote_crdt_map_rr instead of map_rr?

Why then don't name state_orset as lasp_state_orset?

**Comment by cmeiklejohn** [comment]
If you re-read my previous message, I said that the types library is the exception; (types, which, was originally authored by someone that wasn't me) all other libraries use the prefix.

**Comment by bieniusa** [comment]
Well, it doesn't count if you edit your message after my answer 😛

**Comment by cmeiklejohn** [comment]
I hear that concurrency is hard.

**Comment by bieniusa** [comment]
@gyounes Are you ok with re-adding the prefix?

File: src/antidote_crdt_flag_ew.erl [hunk 1]
```diff
@@ -46,38 +46,38 @@
 -define(TAG, 77).
 -define(V1_VERS, 1).
 
--export_type([flag_ew/0]).
--opaque flag_ew() :: flag_helper:flag().
+-export_type([antidote_crdt_flag_ew/0]).
```
Comment (peterzeller) [inline src/antidote_crdt_flag_ew.erl]: I think your search-replace affected too many places, the type names can be kept short ;)

**Pull request merged**